test/types/models.test.ts

```diff
@@ -216,7 +216,7 @@ function find() {
   Project.find({ name: 'Hello' });
 
   // just callback; this is no longer supported on .find()
-  expectError(Project.find((error: CallbackError, result: IProject[]) => console.log(error, result)));
+  Project.find((error: CallbackError, result: IProject[]) => console.log(error, result));
```

Comment:
after the revert, does the "expectError" now fail?

Reply:
Yes. Functions match "object with any keys" in TypeScript, and telling TypeScript "this parameter can be any non-function object" is tricky, I haven't been able to figure out how to do that.

![image](https://github.com/user-attachments/assets/8caa4877-d521-4f3f-88a2-0a6982230bd1)

I would love to keep the "this parameter can't be a function" behavior, but not sure how that would work